tests/integration/framework/kill/kill_posix.go

```diff
@@ -0,0 +1,34 @@
+//go:build !windows
+// +build !windows
+
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kill
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func interrupt(t *testing.T, cmd *exec.Cmd) {
+	// TODO: daprd does not currently gracefully exit on a single interrupt
```

Comment:
Make sure you check the value of graceful-shutdown.

A single kill signal will start the graceful shutdown, and IIRC the default is 5s.

To force a shutdown, send a second kill signal.

Reply:
We are checking the exit code  [here](https://github.com/dapr/dapr/pull/6327/files#diff-5c299386208e2c1d7b4b5257eb6fb95e522ea4f94d6bb0d6a8f67be94f0c416bR213).

It is not really a "graceful" shutdown today, because we are not propagating a shutdown signal to everything properly and waiting for all resources to be released. 5s wait time is not appropriate for these tests, especially when nothing is happening in that 5s time.

When [graceful shutdown](https://github.com/dapr/dapr/issues/2478) is finished, we can get rid of this "hack". This integration testing framework is actually one of the requirements for that work anyway :sweat_smile: